tool/resources/org/antlr/v4/tool/templates/codegen/Python2/Python2.stg

```diff
@@ -448,12 +448,12 @@ TestSetInline(s) ::= <<
 
 // Java language spec 15.19 - shift operators mask operands rather than overflow to 0... need range test
 testShiftInRange(shiftAmount) ::= <<
-((<shiftAmount>) & ~0x3f) == 0
```

Comment:
Not sure about this one, I think I remember a scenario where shiftAmount is an expression, not a value, hence the parenthesis

Reply:
It depends on `offsetShiftVar` that returns expression in parentheses if it's needed. This is a sample:

```py
if ((_la - 3) & ~0x3f) == 0 and ((1 << (_la - 3)) & 9223090579141953573) != 0 or _la==123 or _la==128:
```

Also, `MultiTokenAlternativeDescriptor` test covers all cases of constant folding hence everything works fine.

Reply:
Our test coverage is not great, so I won't rely on them to make this decision, which doesn't optimize anything since a parenthesis expression is a noop

Reply:
I think my test covers all cases. And there are only 2 possible cases for this function: with parentheses and without.

Reply:
Yeah, let's use (...) to guarantee when we're unsure.